src/store/security/security-reducers.ts

```diff
@@ -9,22 +9,25 @@ export const securityReducers = createReducer<SecurityState>(securityInitialStat
   builder.addCase(enterPassword.fail, state => ({
     ...state,
     passwordAttempt: state.passwordAttempt + 1,
-    passwordLockTime: state.passwordAttempt >= MAX_PASSWORD_ATTEMPTS ? Date.now() : state.passwordLockTime
+    passwordLockTime: state.passwordAttempt > MAX_PASSWORD_ATTEMPTS ? Date.now() : state.passwordLockTime
   }));
   builder.addCase(enterPassword.success, state => ({
     ...state,
     passwordAttempt: INITIAL_ENTER_WRONG_PASSWORD_ATTEMPTS
   }));
   builder.addCase(checkApp.submit, state => ({
     ...state,
-    isForceUpdateNeeded: createEntity(state.isForceUpdateNeeded.data, true)
+    isForceUpdateNeeded: createEntity(state.isForceUpdateNeeded.data, true),
+    isAppCheckFailed: createEntity(state.isAppCheckFailed.data, true)
   }));
   builder.addCase(checkApp.success, (state, { payload }) => ({
     ...state,
-    isForceUpdateNeeded: createEntity(payload)
+    isForceUpdateNeeded: createEntity(payload.isForceUpdateNeeded),
+    isAppCheckFailed: createEntity(payload.isAppCheckFailed)
   }));
   builder.addCase(checkApp.fail, (state, { payload }) => ({
     ...state,
-    isForceUpdateNeeded: createEntity(state.isForceUpdateNeeded.data, false, payload)
+    isForceUpdateNeeded: createEntity(state.isForceUpdateNeeded.data, false, payload),
+    isAppCheckFailed: createEntity(state.isAppCheckFailed.data, false, payload)
```

Comment:
maybe let's set `isForceUpdateNeeded.data: false` & `isAppCheckFailed.data: false` is request failed?